docs/docs/dev_docs/contracts/syntax/storage.md

```diff
@@ -1,40 +1,533 @@
+
+#include_code assert_contains_and_remove /yarn-project/aztec-nr/aztec/src/state_vars/set.nr rust
+
+<!---
+@LHerskind
+I don't see why this one is actually needed and could not be deprecated down the line. 
+Allow insertions into the rpc database and this one can be removed.
+-->
+
+### `assert_contains_and_remove_publicly_created`
+
+Like above, this is used to ensure that the message exists in the data tree and then consume it. However, it differs slightly since there is currently a difference between notes that have been inserted from public and private execution. This means that you currently must use this function to consume and nullify a note that was created in a public function. This will be fixed in the future. 
+
+#include_code assert_contains_and_remove_publicly_created /yarn-project/aztec-nr/aztec/src/state_vars/set.nr rust
+
+While this might look intimidating, the use of the function is rather easy, and is used in the following way:
+
+#include_code assert_contains_and_remove_publicly_created /yarn-project/noir-contracts/src/contracts/token_contract/src/main.nr rust
+
+The reason we are not reading this note ahead of time is that no [encrypted log](./events.md#encrypted-events) was emitted for this note, since it was created in public thereby making the encrypted log useless (everyone saw the content ahead of time).
+
+### `remove`
+
+Will remove a note from the set if it previously have been read from storage. That being that you have fetched it through a `get_notes` call. This is useful when you want to remove a note that you have previously read from storage to not have to read it again. If you recall from earlier, we are emitting a nullifier when reading values to make sure that they are up to date. 
+
+#include_code remove /yarn-project/aztec-nr/aztec/src/state_vars/set.nr rust
+
+An example of how to use this operation is visible in the `easy_private_state`:
+
+#include_code remove /yarn-project/aztec-nr/easy-private-state/src/easy_private_state.nr rust
+
+### `get_notes`
+
+This function returns the notes the account has access to:
+
+#include_code get_notes /yarn-project/aztec-nr/aztec/src/state_vars/set.nr rust
+
+There's a limit on the maximum number of notes this function can return at a time. Check [here](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec-nr/aztec/src/constants_gen.nr) and look for `MAX_READ_REQUESTS_PER_CALL` for the up-to-date number.
+
+Because of this limit, we should always consider using the second argument `NoteGetterOptions` to target the notes we need, and to reduce the time required to recursively call this function.
+
+An example of this is is using the [filter_notes_min_sum](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec-nr/value-note/src/filter.nr) to get "enough" notes to cover a given value. 
+
+#include_code get_notes /yarn-project/aztec-nr/easy-private-state/src/easy_private_state.nr rust
+
+### `view_notes`
```

Comment:
Do we use this function anywhere in our source code? I think it would be helpful to see an example.

Reply:
The balance utilities is actually calling this under the hood, will add a reference to it as showcase.